### ISSUE TITLE
DC0004 false positive when [NonDebuggable] attribute is placed before the procedure keyword

### ISSUE THREAD

**Issue by fvet** [ALCops/Analyzers#102]
## DC0004 false positive when `[NonDebuggable]` attribute is placed before the `procedure` keyword

**Rule:** DC0004 — Public procedure requires XML documentation

### Expected behavior

No warning — the procedure has complete XML documentation (`<summary>` + `<param>`).

### Actual behavior

DC0004 is raised despite the XML documentation being present.

### Minimal reproduction

```al
/// <summary>
/// Sets the call body content from a stream.
/// </summary>
/// <param name="ContentStream">The instream containing the call body.</param>
[NonDebuggable]
procedure SetCallBody(ContentStream: InStream)
var
    OutStream: OutStream;
begin
    Clear("Call Body");
    "Call Body".CreateOutStream(OutStream);
    CopyStream(OutStream, ContentStream);
end;
```

### Root cause

`HasXmlDocumentation` checks `method.ProcedureKeyword.LeadingTrivia` for doc comment trivia:

```csharp
var trivia = method.ProcedureKeyword.LeadingTrivia;

return trivia.Any(t =>
    t.Kind == EnumProvider.SyntaxKind.SingleLineDocumentationCommentTrivia ||
    t.Kind == EnumProvider.SyntaxKind.MultiLineDocumentationCommentTrivia);
```

In the AL syntax tree, trivia attaches to the immediately following **token**. When `[NonDebuggable]` is present, the `///` comments become leading trivia of `[` (the attribute's first token) — not of the `procedure` keyword. As a result, `method.ProcedureKeyword.LeadingTrivia` only sees whitespace and no doc comment trivia, causing the rule to fire incorrectly.

### Suggested fix

Check the leading trivia of the first token of the entire method declaration instead:

```csharp
// Instead of:
var trivia = method.ProcedureKeyword.LeadingTrivia;

// Use:
var trivia = method.GetFirstToken().LeadingTrivia;
```

This ensures XML documentation is detected regardless of whether attributes appear before the `procedure` keyword.
